browser_utils.js

```diff
@@ -397,9 +433,12 @@ async function assertNotXPath(page, xpath, options, _timeout=2000, _checkEvery=2
     }
     const {message, timeout, checkEvery} = options;
 
+    const navigation = watchNavigtation(page);
     let remainingTimeout = timeout;
     // eslint-disable-next-line no-constant-condition
     while (true) {
+        await waitForNavigationRequest(page, navigation);
```

Comment:
This imposes an upfront cost for _all_ calls.
Can we instead catch the error and only then do some extra stuff?